aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -135,10 +138,16 @@ public boolean isStopped() {
      * Start performing sync operations between the local storage adapter
      * and the remote GraphQL endpoint.
      */
-    public void start() {
+    public synchronized void start() {
+        LOG.debug("Available permits = " + startStopSemaphore.availablePermits());
+        if (!startStopSemaphore.tryAcquire()) {
```

Comment:
The problem with using acquire is that CRUD operations will then block in `beforeOperation` as threads wait which can have a cascading effect.

Reply:
To build on my last comment, I ended up changing the start method(s) on the orchestrator to return a Completable. This gives the calling more control over timeout and blocking vs non-blocking.

As a result of this change, the logic to trigger the DataStore's READY event is now triggered from the plugin instead of the orchestrator, which seems appropriate.

Reply:
I removed the overloads that had timeouts since I was able to achieve the same result by re-working the construction of the completable.